addons/website_mail/static/src/js/follow.js

```diff
@@ -4,25 +4,41 @@ odoo.define('website_mail.follow', function (require) {
 var publicWidget = require('web.public.widget');
 
 publicWidget.registry.follow = publicWidget.Widget.extend({
-    selector: '.js_follow',
+    selector: '#wrapwrap:has(.js_follow)',
     disabledInEditableMode: false,
 
+    /**
+     * @override
+     */
     start: function () {
         var self = this;
         this.is_user = false;
+        var $jsFollowEls = this.$el.find('.js_follow');
 
         var always = function (data) {
-            self.is_user = data.is_user;
-            self.email = data.email;
-            self.toggle_subscription(data.is_follower, data.email);
-            self.$target.removeClass('d-none');
+            self.is_user = data[0].is_user;
+            var $jsFollowToEnable = $();
+            _.each(data[1], (resIds, model) => _.each(resIds, (resId) => {
+                $jsFollowToEnable = $jsFollowToEnable.add(
+                    $jsFollowEls.filter('[data-id="' + resId + '"][data-object="' + model + '"]')
```

Comment:
This is the part I was refering.
This basically loops through RPC results (that only contains followed ones from the ones send in RPC) to add them to a variable so we can then call `_toggle_subscription()` in one call (through `_enable_subscription()`).
This could be removed so there is no diff on subscription methods and just call it directly in the loop.

The think is if you have 5.000 tags and are follower of 4000 tags, this code will do 1 call to `_toggle_subscription(true)` and 1000 calls to `_toggle_subscription(false)`.
If we revert it, it will be 5000 calls to `_toggle_subscription()`.
But not sure of the real gain.